### PR TITLE
[sdk] screen orientation tests

### PR DESCRIPTION
**Pull request by quinlanj**
# Why

ScreenOrientation testing portion

Related PRS:
Docs - https://github.com/expo/universe/pull/3280
JS - https://github.com/expo/expo/pull/3103
iOS - https://github.com/expo/expo/pull/3104
Android - https://github.com/expo/expo/pull/3105
Tests - https://github.com/expo/expo/pull/3106
Previous Feedback (1st pass android) - https://github.com/expo/expo/pull/2832

### CODE REVIEW

**Review by esamelson — COMMENTED** [review]
lgtm!

File: apps/test-suite/tests/ScreenOrientation.js [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+            ScreenOrientation.OrientationLock.LANDSCAPE_LEFT,
+            ScreenOrientation.Orientation.PORTRAIT_UP,
+          ];
+          await applyAsync({ desiredOrientations, validOrientations });
+        } catch (error) {
+          t.fail(error);
+        }
+      });
+
+      t.it('Apply a native ios lock', async () => {
```
Comment (ide) [inline apps/test-suite/tests/ScreenOrientation.js]:
```suggestion
      t.it('Apply a native iOS lock', async () => {
```

File: apps/test-suite/tests/ScreenOrientation.js [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+        } catch (error) {
+          t.fail(error);
+        }
+      });
+
+      t.it('Remove all listeners and expect them never to be called', async () => {
+        try {
+          // Register for screen orientation changes
+          let listenerWasCalled = false;
+          ScreenOrientation.addOrientationChangeListener(async () => {
```
Comment (ide) [inline apps/test-suite/tests/ScreenOrientation.js]:
```suggestion
          ScreenOrientation.addOrientationChangeListener(() => {
```

etc...

In most event listener APIs, the listener system is not set up to accept async functions (errors go uncaught)

File: apps/test-suite/tests/ScreenOrientation.js [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+      });
+
+      t.it('ensure that we correctly detect our supported orientationLocks', async () => {
+        // orientation locks that we should be able to apply
+        const acceptedLocks = [
+          ScreenOrientation.OrientationLock.ALL,
+          ScreenOrientation.OrientationLock.LANDSCAPE_RIGHT,
+        ];
+
+        for (let lock of acceptedLocks) {
```
Comment (ide) [inline apps/test-suite/tests/ScreenOrientation.js]:
```suggestion
        for (const lock of acceptedLocks) {
```
consistency

**Review by ide — CHANGES_REQUESTED** [review]

File: apps/test-suite/tests/ScreenOrientation.js (hunk 2 above)
Comment (quinlanj) [reply apps/test-suite/tests/ScreenOrientation.js]: ive gone ahead and applied this to all the other cases

**Pull request merged**